pkg/utils/kubernetes/health/health.go

```diff
@@ -143,7 +143,7 @@ func CheckStatefulSet(statefulSet *appsv1.StatefulSet) error {
 // CheckEtcd checks whether the given Etcd is healthy.
 // A Etcd is considered healthy if its ready field in status is true.
 func CheckEtcd(etcd *druidv1alpha1.Etcd) error {
-	if !etcd.Status.Ready {
+	if etcd.Status.Ready == nil || !*etcd.Status.Ready {
```

Comment:
Nit

```suggestion
			case !utils.IsTrue(etcd.Status.Ready):
```

with `"github.com/gardener/gardener/pkg/utils"`